public/react-services/error-orchestrator/error-orchestrator-base.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Wazuh app - Error Orchestrator base
+ * Copyright (C) 2015-2021 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import { ErrorToastOptions } from 'kibana/public';
+import { ErrorOrchestrator, LoggerLevels, UIErrorLog } from '../../../common/constants';
+import { getToasts } from '../../kibana-services';
+import loglevel from 'loglevel';
+import { GenericRequest } from '../../react-services/generic-request';
+
+export class ErrorOrchestratorBase implements ErrorOrchestrator {
+  public loadErrorLog(errorLog: UIErrorLog) {
+    if (errorLog.display) this.displayError(errorLog);
+    if (errorLog.store) this.storeError(errorLog).then(loglevel.info);
+  }
+
+  public displayError(errorLog: UIErrorLog) {
+    getToasts().addError(errorLog.error.error, {
+      title: errorLog.error.title,
+    } as ErrorToastOptions);
+  }
+
+  public loglevelError(errorLog: UIErrorLog) {
+    switch (errorLog.level) {
+      case LoggerLevels.INFO:
+        loglevel.info(errorLog.error.message, errorLog.error.error);
+        break;
+      case LoggerLevels.WARNING:
+        loglevel.warn(errorLog.error.message, errorLog.error.error);
+        break;
+      case LoggerLevels.ERROR:
+        loglevel.error(errorLog.error.message, errorLog.error.error);
+        break;
+      default:
+        console.log('No error level', errorLog.error.message, errorLog.error.error);
+    }
+  }
+
+  private async storeError(errorLog: UIErrorLog) {
+    await GenericRequest.request('POST', `/utils/logs/ui`, {
+      body: {
+        message: errorLog.error.message,
+        level: errorLog.level,
+        location: errorLog.location,
+      },
+    })
+      .then(loglevel.info)
+      .catch(loglevel.error);
```

Comment:
Maybe this could be refactored to use `try/catch` block instead `.then/.catch` to manage the Promise.

Reply:
Absolutely, working on that! thx